fix: [nvbug5300494] Use runtime total gpu memory to calculate kv cache memory and log more memory information

HuiGao-NV opened the pull request:
Use runtime total gpu memory to calculate kv cache memory and log more memory information.
This can avoid mis-reporting of less kv memory.

tensorrt-cicd commented:
[PR_Github #6467](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6467/) [ run ] completed with state `SUCCESS`
[/LLM/release-0.20/L0_MergeRequest_PR pipeline #70](https://nv/trt-llm-cicd/job/release-0.20/job/L0_MergeRequest_PR/70/) completed with status: 'FAILURE'

HuiGao-NV commented:
/bot run

tensorrt-cicd commented:
[PR_Github #6510](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6510/) [ run ] triggered by Bot

HuiGao-NV commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #6514](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6514/) [ run ] triggered by Bot

tensorrt-cicd commented:
[PR_Github #6514](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6514/) [ run ] completed with state `SUCCESS`
[/LLM/release-0.20/L0_MergeRequest_PR pipeline #75](https://nv/trt-llm-cicd/job/release-0.20/job/L0_MergeRequest_PR/75/) completed with status: 'FAILURE'

HuiGao-NV commented:
/bot run --disable-fail-fast

tensorrt-cicd commented:
[PR_Github #6586](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6586/) [ run ] completed with state `SUCCESS`
[/LLM/release-0.20/L0_MergeRequest_PR pipeline #82](https://nv/trt-llm-cicd/job/release-0.20/job/L0_MergeRequest_PR/82/) completed with status: 'FAILURE'

tensorrt-cicd commented:
[PR_Github #6651](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/6651/) [ run ] completed with state `SUCCESS`
[/LLM/release-0.20/L0_MergeRequest_PR pipeline #87 `(Partly Tested)`](https://nv/trt-llm-cicd/job/release-0.20/job/L0_MergeRequest_PR/87/) completed with status: 'SUCCESS'

HuiGao-NV committed:
Log more memory information
Change method to compute peak memory
Set new peak memory for case test_ptq_quickstart_advanced_mtp
Get non-torch memory of starttime of kv memory estimation

Signed-off-by: Hui Gao <huig@nvidia.com>

HuiGao-NV committed:
Merge branch 'release/0.20' into oom_testing
